src/tests/GeoTest.cpp

```diff
@@ -98,3 +99,64 @@ TEST(GeoTest, CoordinateFrame) {
   c4.fromJson(j);
   EXPECT_EQ(c3, c4);
 }
+
+// standard method
+// transform the 8 corners, and extract the min and max
+Magnum::Range3D getTransformedBB_standard(const Magnum::Range3D& range,
+                                          const Magnum::Matrix4& xform) {
+  std::vector<Magnum::Vector3> corners;
+  corners.push_back(xform.transformPoint(range.frontBottomLeft()));
+  corners.push_back(xform.transformPoint(range.frontBottomRight()));
+  corners.push_back(xform.transformPoint(range.frontTopLeft()));
+  corners.push_back(xform.transformPoint(range.frontTopRight()));
+
+  corners.push_back(xform.transformPoint(range.backTopLeft()));
+  corners.push_back(xform.transformPoint(range.backTopRight()));
+  corners.push_back(xform.transformPoint(range.backBottomLeft()));
+  corners.push_back(xform.transformPoint(range.backBottomRight()));
+
+  Magnum::Range3D transformedBB{Magnum::Math::minmax<Magnum::Vector3>(corners)};
+
+  return transformedBB;
+}
+
+TEST(GeoTest, Aabb) {
+  // Generate N boxes in random position and orientation.
+  // compute aabb for each box using standard method and library method
+  // respectively.
+  // compare the results, which should be identical
+
+  Magnum::Range3D box = Magnum::Range3D{Magnum::Vector3{-10.0, -10.0, -10.0},
+                                        Magnum::Vector3{10.0, 10.0, 10.0}};
+
+  int numBoxes = 100;
+  for (int iBox = 0; iBox < numBoxes; ++iBox) {
+    // generate random translation
+    const Magnum::Vector3 translation{
+        static_cast<float>(rand() % 1000) + (rand() % 1000) / 1000.0f,
+        static_cast<float>(rand() % 1000) + (rand() % 1000) / 1000.0f,
+        static_cast<float>(rand() % 1000) + (rand() % 1000) / 1000.0f};
+
+    // generate random rotation using:
+    // http://planning.cs.uiuc.edu/node198.html
+    double u1 = (rand() % 1000) / 1000.0;
+    double u2 = (rand() % 1000) / 1000.0;
+    double u3 = (rand() % 1000) / 1000.0;
+
+    Magnum::Vector3 qAxis(sqrt(1 - u1) * cos(2 * M_PI * u2),
+                          sqrt(u1) * sin(2 * M_PI * u3),
+                          sqrt(u1) * cos(2 * M_PI * u3));
+
+    // assemble the transformation matrix
+    Magnum::Matrix4 xform = Magnum::Matrix4::from(
+        Magnum::Quaternion(qAxis, sqrt(1 - u1) * sin(2 * M_PI * u2)).toMatrix(),
```

Comment:
I remember @aclegg3 added a similar piece of code here: https://github.com/facebookresearch/habitat-sim/blob/f2845b58217883517586feaad18fdcb46180f511/src/utils/viewer/viewer.cpp#L278-L289 ... maybe turn it into an utility function?

This would be great to have in `Magnum::Math` of course, I'll think about adding it there.

Reply:
Yes, that was where I copied ("duplicated") the code. Sorry, I am lazy. Definitely we need to put it into a utility file. :-)